app/Http/Controllers/AccessoriesController.php

```diff
@@ -327,33 +369,55 @@ public function postCheckin(Request $request, $accessoryUserId = null, $backto =
-            if (!is_null($accessory_user->assigned_to)) {
-                $user = User::find($accessory_user->assigned_to);
+        // Quantity check
+        $max_to_checkin = $accessory_user->assigned_qty;
+        if ($max_to_checkin <= 0) {
+            return redirect()->route('accessories.index')->with('error', trans('admin/accessories/message.checkin.error'));
+        }
+        $validator = Validator::make($request->all(), [
+            "qty" => "required|numeric|between:1,$max_to_checkin"
+        ]);
+        if ($validator->fails()) {
+            return redirect()->back()
+                ->withErrors($validator)
+                ->withInput();
+        }
+        // Quantity check was successful
+
+        // Quantity adjust
+        $checkin_qty = (int)request('qty');
+        $qty_remaining_in_checkout = ($accessory_user->assigned_qty - $checkin_qty);
+        if ($qty_remaining_in_checkout > 0) {
+            // Update to correct checked out quantity
+            DB::table('accessories_users')->where('id', $accessoryUserId)->update(['assigned_qty' => $qty_remaining_in_checkout]);
+        } else {
+            if (is_null(DB::table('accessories_users')->where('id', '=', $accessory_user->id)->delete())) {
+                return redirect()->route('accessories.index')->with('error', trans('admin/accessories/message.checkin.error'));
             }
+        }
 
-            $data['log_id'] = $logaction->id;
-            $data['first_name'] = e($user->first_name);
-            $data['last_name'] = e($user->last_name);
-            $data['item_name'] = e($accessory->name);
-            $data['checkin_date'] = e($logaction->created_at);
-            $data['item_tag'] = '';
-            $data['note'] = e($logaction->note);
 
-            if ($backto=='user') {
-                return redirect()->route("users.show", $return_to)->with('success', trans('admin/accessories/message.checkin.success'));
-            }
-            return redirect()->route("accessories.show", $accessory->id)->with('success', trans('admin/accessories/message.checkin.success'));
+        $accessory->logCheckin($assigned_user, e(request('note')));
+
+        // Unused anywhere ?, should delete ?
+        /*
+        $data['log_id'] = $logaction->id;
+        $data['first_name'] = e($assigned_user->first_name);
+        $data['last_name'] = e($assigned_user->last_name);
+        $data['item_name'] = e($accessory->name);
+        $data['checkin_date'] = e($logaction->created_at);
+        $data['item_tag'] = '';
+        $data['note'] = e($logaction->note);
+        */
+
+        if ($backto=='user') {
+            return redirect()->route("users.show", $assigned_user->id)->with('success', trans('admin/accessories/message.checkin.success'));
         }
-        // Redirect to the accessory management page with error
-        return redirect()->route('accessories.index')->with('error', trans('admin/accessories/message.checkin.error'));
```

Comment:
I'm assuming that this is handled in the new version up a little bit higher - line 399 - 

https://github.com/snipe/snipe-it/pull/7849/files#diff-a75f1db0ffa83e9a7eb4258a696621c491b693d3d9025da52f2f8845da056fd0R399